packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js

```diff
@@ -0,0 +1,109 @@
+import { DKIM_CONFIGURATION_GUIDE } from './emailpro-domain-dkim-autoconfig.constants';
+
+export default class EmailProDomainDkimAutoconfigCtrl {
+  /* @ngInject */
+  constructor(
+    $q,
+    $scope,
+    $stateParams,
+    $translate,
+    coreConfig,
+    EmailPro,
+    EmailProDomains,
+  ) {
+    this.services = {
+      $q,
+      $scope,
+      $stateParams,
+      $translate,
+      EmailPro,
+      EmailProDomains,
+    };
+
+    this.domain = $scope.currentActionData.domain;
+    this.dkimStatus = $scope.currentActionData.dkimStatus;
+    this.GLOBAL_DKIM_STATUS =
+      $scope.currentActionData.constant.GLOBAL_DKIM_STATUS;
+    this.services.$scope.configDkim = () => this.configDkim();
+
+    this.dkimGuideLink =
+      DKIM_CONFIGURATION_GUIDE[coreConfig.getUser().ovhSubsidiary] ||
+      DKIM_CONFIGURATION_GUIDE.DEFAULT;
+
+    this.init();
```

Comment:
use $onInit to make the construction lighter

Reply:
In fact I need the result in the constructor the $onInit is played too late. If I do a $onInit I break the data-wizard button management.